### ISSUE TITLE
Error when reading from DANDI embargo files

### ISSUE THREAD

**Issue by luiztauffer** [magland/remfile#4]
@magland I'm trying to read an embargo file with remfile using the method we discussed [here](https://github.com/scratchrealm/neurobass/issues/28)

```python
import h5py
import remfile
import requests

url = "https://api.dandiarchive.org/api/assets/f2b7b613-2403-4c7e-b531-935dd97decef/download/"
headers = {
    "Authorization": 'token MY_TOKEN'
}
response = requests.get(url, headers=headers, stream=True)
authorized_url = response.url

file = remfile.File(authorized_url)

f = h5py.File(file, 'r')
```

but I'm getting this error:
```python
---> 12 file = remfile.File(authorized_url)
     14 f = h5py.File(file, 'r')

File /media/luiz/storage2/conda_environments/env_shuler/lib/python3.8/site-packages/remfile/RemFile.py:56, in RemFile.__init__(self, url, verbose, disk_cache, _min_chunk_size, _max_cache_size, _chunk_increment_factor, _bytes_per_thread, _max_threads, _max_chunk_size, _impose_request_failures_for_testing)
     54 self._smart_loader_chunk_sequence_length = 1
     55 response = requests.head(self._url)
---> 56 self.length = int(response.headers['Content-Length'])

File /media/luiz/storage2/conda_environments/env_shuler/lib/python3.8/site-packages/requests/structures.py:52, in CaseInsensitiveDict.__getitem__(self, key)
     51 def __getitem__(self, key):
---> 52     return self._store[key.lower()][1]

KeyError: 'content-length'
```

**Comment by magland** [comment]
It seems that content-length is not being included in the response header (remfile is trying to determine the total size). Could you try the following

```
curl -I "URL"
```

where URL is authorized_url

**Comment by magland** [comment]
I just tried this and confirmed that it is not returning the content-length. Let me look at the remfile source code to see if it really needs to know the size.

Note: in the CURL command, you need to use quotes around the URL.

**Comment by magland** [comment]
Sorry, I was going down the wrong route. I think the issue is that HEAD request is not allowed for the authorized URL (only GET). So I can try to modify remfile to only use GET request to get the total file size...

**Comment by magland** [comment]
@luiztauffer This should be resolved with #6
This has been published to pypi as version 0.1.7

**Comment by luiztauffer** [comment]
the problem is solved in the new version, thanks!
